packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -599,6 +603,10 @@ export const Table = <
     setPageSize(pageSize);
   }, [pageSize, setPageSize]);
 
+  const numSelectedRows =
+    selectionMode === 'single'
+      ? 0
+      : rows.filter((row) => row.isSelected).length;
```

Comment:
Are we also counting subrows in this `numSelectedRows` count?

If so, then you might need to use `flatRows`. It is an ordered array of all rows (even subrows).